client/client_test.go

```diff
@@ -215,3 +215,37 @@ func TestGetActionTimeout(t *testing.T) {
 		})
 	}
 }
+
+func Test_argListFromPgParams(t *testing.T) {
+	type testParams struct {
+		req      *agentpb.StartActionRequest_PTPgSummaryParams
+		expected []string
+	}
+
+	testCases := []*testParams{
+		{&agentpb.StartActionRequest_PTPgSummaryParams{Host: "10.20.30.40", Port: 555, Username: "person",
+			Password: "secret"}, []string{"--host", "10.20.30.40", "--port", "555", "--username", "person", "--password", "secret"}},
+		{&agentpb.StartActionRequest_PTPgSummaryParams{Host: "10.20.30.40", Port: 555, Username: "person",
+			Password: ""}, []string{"--host", "10.20.30.40", "--port", "555", "--username", "person"}},
+		{&agentpb.StartActionRequest_PTPgSummaryParams{Host: "10.20.30.40", Port: 555, Username: "",
+			Password: "secret"}, []string{"--host", "10.20.30.40", "--port", "555", "--password", "secret"}},
+		{&agentpb.StartActionRequest_PTPgSummaryParams{Host: "10.20.30.40", Port: 65536, Username: "",
+			Password: "secret"}, []string{"--host", "10.20.30.40", "--password", "secret"}},
+		{&agentpb.StartActionRequest_PTPgSummaryParams{Host: "", Port: 555, Username: "", Password: "secret"},
+			[]string{"--port", "555", "--password", "secret"}},
+
+		{&agentpb.StartActionRequest_PTPgSummaryParams{Host: "", Port: 0, Username: "", Password: ""}, []string{}},
+		{&agentpb.StartActionRequest_PTPgSummaryParams{Host: "", Port: 0, Username: "王华", Password: `"`},
+			[]string{"--username", "王华", "--password", `"`}},
+		{&agentpb.StartActionRequest_PTPgSummaryParams{Host: "10.20.30.40", Port: 555, Username: "",
+			Password: "   "}, []string{"--host", "10.20.30.40", "--port", "555", "--password", "   "}},
```

Comment:
it should not work this way, if we will have empty password we should pass flag.

Reply:
What do you want to do here? If there is a whitespace 0x20 within the password, don't pass the password? Or strip the spaces from left and right and if it becomes "", not to pass the password?

Reply:
yes, let's trim spaces from left and right and it becomes empty string do not pass the password.

Reply:
What about this?